Update docs regarding the native config and logging config (#788)
wso2/docs-choreo-dev: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md

During deployment, Choreo performs a Trivy scan to detect critical vulnerabilities in third-party libraries added to the integration component. If the scan detects any critical vulnerabilities, it halts the deployment process. The deployment pane shows the Trivy scan status and any security failures in the Library (Trivy) vulnerable scan step, which you can access by clicking on it. After fixing the vulnerability, you can redeploy the component.

## Configuring Micro Integrator

> WSO2 MI comes with a set of pre-configured settings by default, designed to work in a general context. However, these default configurations might not be suitable for every organization's specific needs. As a result, it is crucial to modify the default configuration in WSO2 MI to optimize its performance and ensure compatibility with an organization's systems and applications.

Reviewer: ```suggestion
WSO2 Micro Integrator (WSO2 MI) comes with preconfigured settings to work in a general context. However, these settings may not apply to the specific requirements of an organization. Therefore, it is important to customize the WSO2 MI preconfigured settings to optimize its performance and ensure compatibility with the systems and applications of the organization.
```

> To customize the default configurations in WSO2 MI, users can apply their desired configurations to the instances running on Choreo. To add new configurations, users need to define a `deployment.toml` file in the project sub-path. Similar to configuring MI locally, users can apply configuration changes in the `deployment.toml` file. However, it is essential to apply only the required changes to avoid disrupting internal communication. It's worth noting that altering default configurations such as port offset, hostname, and other critical settings might break internal communication. Therefore, it's advisable to avoid modifying them unless necessary.

Reviewer: When we say "To add new configurations, users need to define a deployment.toml file in the project sub-path", I feel Choreo users might not clearly understand where to define the deployment.toml? 
Should we add more clarity here?

Author: ack

> Below is a sample `deployment.toml` configuration for configuring JMS transport. Refer to

Reviewer: Suggestion:

Here is a sample `deployment.toml` file that can be used to configure the JMS transport. For more information on WSO2 MI configuration parameters, see the [MI Config Catalog](https://apim.docs.wso2.com/en/latest/reference/config-catalog-mi/).

During deployment, Choreo performs a Trivy scan to detect critical vulnerabilities in third-party libraries added to the integration component. If the scan detects any critical vulnerabilities, it halts the deployment process. The deployment pane shows the Trivy scan status and any security failures in the Library (Trivy) vulnerable scan step, which you can access by clicking on it. After fixing the vulnerability, you can redeploy the component.

## Configuring Micro Integrator

By default, WSO2 MI comes with a set of pre-configured settings that are designed to work in a general context. However,
these default configurations may not be suitable for every organization's specific needs. Therefore, it is important to
change the default configuration in WSO2 MI to optimize its performance and ensure its compatibility with the
organization's systems and applications.

> To modify the default configurations in WSO2 MI, users can apply their desired configurations in the instances that runs
> on the Choreo. To add new configurations, users need to define a `deployment.toml` file in the project sub-path.
> Similar to configuring MI locally, users can apply configuration changes in the `deployment.toml`. However, it's
> essential to apply only the required changes to avoid breaking the internal communication. It's worth noting that
> changing default configurations like port offset, hostname, and other critical settings might break the internal
> communication. Therefore, it's advisable to avoid modifying them unless it's necessary.

Reviewer 2: ```suggestion
To customize the default configurations in WSO2 MI, users can apply their desired configurations to the instances running on Choreo. To add new configurations, users need to define a `deployment.toml` file in the project sub-path. Similar to configuring MI locally, users can apply configuration changes in the `deployment.toml` file. However, it is essential to apply only the required changes to avoid disrupting internal communication. It's worth noting that altering default configurations such as port offset, hostname, and other critical settings might break internal communication. Therefore, it's advisable to avoid modifying them unless necessary.
```

> A sample `deployment.toml` configuration to configure JMS transport is looks like below. Also refer to
> the [MI Config Catalog](https://apim.docs.wso2.com/en/latest/reference/config-catalog-mi/) to know more about configuring
> Micro Integrator.

Reviewer 2: ```suggestion
Below is a sample `deployment.toml` configuration for configuring JMS transport. Refer to
the [MI Config Catalog](https://apim.docs.wso2.com/en/latest/reference/config-catalog-mi/) for more information about configuring the Micro Integrator.
```

For a comprehensive list of parameters that can be configured as environment variables, please refer to the [WSO2 API Manager Documentation - Injecting Parameters - Supported parameters](https://apim.docs.wso2.com/en/latest/integrate/develop/injecting-parameters/#supported-parameters).

## Configuring logging

> Proper logging mechanisms are crucial when developing and maintaining a software system. Logging captures events and messages that occur during an application's execution, providing valuable information for troubleshooting and debugging purposes.

Reviewer: ```suggestion
When you develop and maintain a software system, it is crucial to implement proper logging mechanisms. Proper logging can save time and effort during the development, testing, and maintenance phases of the software development process.

Logging captures events and messages that occur during the execution of an application. These events and messages provide insights to troubleshoot and debug issues.
```

> In Micro Integrator (MI) instances, logging can be configured and customized to meet specific requirements. The logging

Reviewer: Suggestion:
In Micro Integrator instances, you can configure and customize logging depending on your requirement. You can add logging configurations to each MI instance's environment, which allows you to fine-tune logging depending on the specific environment or deployment scenario.

To configure logging in MI instances, you can use environment variables with specific naming conventions as follows:
1. Start the variable name with `logging_level_` followed by the package or class name.
2. Replace the dot character in the package name with an underscore.
3. Set the variable value to the required logging level for the corresponding package or class.

For example, to set the logging level for the `org.apache.synapse.transport.http.wire` package to `debug`, define the following environment variable:

```
export logging_level_org_apache_synapse_transport_http_wire=debug
```

This approach allows you to control the logging granularity, which helps you troubleshoot and identify issues faster.

Author: ack

For a comprehensive list of parameters that can be configured as environment variables, please refer to the [WSO2 API Manager Documentation - Injecting Parameters - Supported parameters](https://apim.docs.wso2.com/en/latest/integrate/develop/injecting-parameters/#supported-parameters).

## Configuring logging

> When developing and maintaining a software system, one of the crucial aspects is to have proper logging mechanisms in
> place. Logging is a way to capture events and messages that occur during the execution of an application, providing
> valuable information for troubleshooting and debugging purposes.

Reviewer 2: ```suggestion
Proper logging mechanisms are crucial when developing and maintaining a software system. Logging captures events and messages that occur during an application's execution, providing valuable information for troubleshooting and debugging purposes.
```
